scipy/interpolate/_polyint.py

```diff
@@ -541,33 +563,74 @@ class BarycentricInterpolator(_Interpolator1D):
 
     Based on Berrut and Trefethen 2004, "Barycentric Lagrange Interpolation".
 
+    Examples
+    --------
+    To produce a quintic barycentric interpolant approximating the function
+    :math:`\sin x`, and its first four derivatives, using six randomly-spaced
+    nodes in :math:`(0, \frac{\pi}{2})`:
+
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.interpolate import BarycentricInterpolator
+    >>> xi = np.random.rand(6) * np.pi/2
+    >>> f, f_d1, f_d2, f_d3, f_d4 = np.sin, np.cos, lambda x: -np.sin(x), lambda x: -np.cos(x), np.sin
+    >>> P = BarycentricInterpolator(xi, f(xi))
+    >>> fig, axs = plt.subplots(5, 1, sharex=True, layout='constrained', figsize=(6,10))
+    >>> x = np.linspace(0, np.pi, 100)
+    >>> axs[0].plot(x, f(x), 'k--', xi, f(xi), '.k', x, P(x), 'r:')
+    >>> axs[1].plot(x, f_d1(x), 'k--', xi, f_d1(xi), '.k', x, P.derivative(x), 'r:')
+    >>> axs[2].plot(x, f_d2(x), 'k--', xi, f_d2(xi), '.k', x, P.derivative(x, 2), 'r:')
+    >>> axs[3].plot(x, f_d3(x), 'k--', xi, f_d3(xi), '.k', x, P.derivative(x, 3), 'r:')
+    >>> axs[3].plot(x, f_d3(x), 'k--', xi, f_d3(xi), '.k', x, P.derivative(x, 3), 'r:')
+    >>> axs[4].plot(x, f_d4(x), 'k--', xi, f_d4(xi), '.k', x, P.derivative(x, 4), 'r:')
+    >>> axs[0].set_xlim(0, np.pi)
+    >>> axs[4].set_xlabel(r"$x$")
+    >>> axs[0].set_ylabel("$f(x)$")
+    >>> axs[1].set_ylabel("$f'(x)$")
+    >>> axs[2].set_ylabel("$f''(x)$")
+    >>> axs[3].set_ylabel("$f^{(3)}(x)$")
+    >>> axs[4].set_ylabel("$f^{(4)}(x)$")
+    >>> labels = ['True function $f$', 'Interpolation nodes', 'Barycentric interpolation']
+    >>> axs[0].legend(axs[0].get_lines(), labels, bbox_to_anchor=(0., 1.02, 1., .102),
+    ...               loc='lower left', ncols=3, mode="expand", borderaxespad=0., frameon=False)
+    >>> plt.tight_layout()
+    >>> plt.show()
```

Comment:
![image](https://user-images.githubusercontent.com/60778417/227737478-e55a81ac-46b5-4e96-a9d6-726c2978d572.png)
Can you make the nodes easier to see against the lines. Also minor nit but the text in the legend overlaps